### PR TITLE
[UTOPIA-360] updated PIA endpoints with newly added CPO role

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- updated PIA endpoints with newly added CPO role

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Added tests. Created new Role in SSO and assigned to myself to verify

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [x] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/backend/src/modules/pia-intake/pia-intake.service.ts [hunk 1]
```diff
@@ -199,6 +202,15 @@ export class PiaIntakeService {
         status: Not(PiaIntakeStatusEnum.INCOMPLETE),
       });
     }
+
+    // Scenario 3: As a CPO, retrieve all pia-intakes in CPO_Review
+    if (isCPO) {
+      whereClause.push({
+        ...commonWhereClause,
+        status: PiaIntakeStatusEnum.PCT_REVIEW,
```
Comment (Bottle7) [inline src/backend/src/modules/pia-intake/pia-intake.service.ts:210]:
```suggestion
        status: PiaIntakeStatusEnum.CPO_REVIEW,
```

File: src/backend/src/modules/pia-intake/pia-intake.service.ts [hunk 2]
```diff
@@ -422,6 +439,12 @@ export class PiaIntakeService {
       return true;
     }
 
+    // Scenario 3: PIA is in CPO_REVIEW status
+    const isCPO = this.isCPO(userRoles);
+    if (isCPO && piaIntake.status === PiaIntakeStatusEnum.PCT_REVIEW) {
```
Comment (Bottle7) [inline src/backend/src/modules/pia-intake/pia-intake.service.ts:444]:
```suggestion
    if (isCPO && piaIntake.status === PiaIntakeStatusEnum.CPO_REVIEW) {
```

File: src/backend/test/unit/pia-intake/pia-intake.service.spec.ts [hunk 3]
```diff
@@ -1309,6 +1315,115 @@ describe('PiaIntakeService', () => {
...
+
+      expect(piaIntakeRepository.findAndCount).toHaveBeenCalledWith({
+        where: [
+          {
+            isActive: true,
+            createdByGuid: user.idir_user_guid,
+          },
+          {
+            isActive: true,
+            status: PiaIntakeStatusEnum.PCT_REVIEW,
```
Comment (Bottle7) [inline src/backend/test/unit/pia-intake/pia-intake.service.spec.ts:1349]:
```suggestion
            status: PiaIntakeStatusEnum.CPO_REVIEW,
```

File: src/backend/test/unit/pia-intake/pia-intake.service.spec.ts [hunk 4]
```diff
@@ -1309,6 +1315,115 @@ describe('PiaIntakeService', () => {
...
+            createdByGuid: user.idir_user_guid,
+          },
+          {
+            isActive: true,
+            ministry: null, // from spy
+            status: Not(PiaIntakeStatusEnum.INCOMPLETE),
+          },
+          {
+            isActive: true,
+            status: PiaIntakeStatusEnum.PCT_REVIEW,
```
Comment (Bottle7) [inline src/backend/test/unit/pia-intake/pia-intake.service.spec.ts:1407]:
```suggestion
            status: PiaIntakeStatusEnum.CPO_REVIEW,
```

**Comment by Bottle7** [comment]
I think you will want to update the PiaStatusesEnum as well.

**Comment by kushal-arora-fw** [comment]
> I think you will want to update the PiaStatusesEnum as well.

Good catch. I was under the impression that it will be updated as part of the other ticket. But it now makes sense now that you have added raised PR for FE changes. Also, added a migration for existing data.

**Review by Bottle7 — APPROVED** [review]

**Pull request merged**